README.md

```diff
@@ -46,7 +249,48 @@ async () => {
 }
 ```
 
-### Consuming messages with consumer groups
+#### <a name="producing-messages-custom-partitioner"></a> Custom partitioner
+
+It's possible to assign a custom partitioner to the consumer. A partitioner is a function which returns another function responsible for the partition selection, something like this:
+
+```javascript
+const MyPartitioner = () => {
+  // some initialization
+  return ({ topic, partitionMetadata, message }) => {
+    // select a partition based on some logic
+    // return the partition number
+    return 0
+  }
+}
+```
+
+`partitionMetadata` is an array of partitions with the following structure:
+
+```javascript
+[
+  { partitionId: 1, leader: 1 },
```

Comment:
Maybe include the signature of this. I would presume that `leader` is the node id of a broker?

Reply:
done